Makefile

```diff
@@ -405,13 +402,13 @@ prepare-coverage-test: $(GOTESTSUM) $(TEST_OUTPUT_ROOT)
 
 unit-test-coverage: prepare-coverage-test
 	@printf $(COLOR) "Run unit tests with coverage..."
-	$(GOTESTSUM) --rerun-fails=$(FAILED_TEST_RETRIES) --rerun-fails-max-failures=10 --junitfile $(NEW_REPORT) --packages $(UNIT_TEST_DIRS) -- \
+	go run ./cmd/tools/test-runner $(GOTESTSUM) -retries $(FAILED_TEST_RETRIES) --junitfile $(NEW_REPORT) --packages $(UNIT_TEST_DIRS) -- \
```

Comment:
I think there's a way to pre-build this in our CI setup. I can look into that if anyone thinks using `go run` would be an issue.

Reply:
Anything you build in `pre-build-functional-test-coverage` will go into the per-run cache. That won't cover the final binary (I think) but the object files. I wouldn't worry about it, though, the compile+link time should be negligible.